packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/CallbackExtensions.cs

```diff
@@ -41,28 +42,29 @@ public static object InvokeCallback(this Callback callback, IReferenceMap refere
             }
         }
 
-        static object InvokeCallbackCore(this Callback callback, IReferenceMap referenceMap)
+        static object InvokeCallbackCore(this Callback callback, IReferenceMap referenceMap, out TypeReference returnType)
```

Comment:
I would prefer to have a concrete structure of { type, value } than using out variables. This makes code hard to follow.

Reply:
I was adopting the surrounding style here, but yeah, adding a `CallbackResult` type.